pyproject.toml

```diff
@@ -175,9 +177,18 @@ before-build = [
 HATCH_BUILD_HOOKS_ENABLE = "1"
 MYPYC_OPT_LEVEL = "3"
 MYPYC_DEBUG_LEVEL = "0"
+AIOHTTP_NO_EXTENSIONS = "1"
+
 # Black needs Clang to compile successfully on Linux.
 CC = "clang"
-AIOHTTP_NO_EXTENSIONS = "1"
+
+[tool.cibuildwheel.macos]
+build-frontend = { name = "build", args = ["--no-isolation"] }
+# Unfortunately, hatch doesn't respect MACOSX_DEPLOYMENT_TARGET
+before-build = [
+    "python -m pip install 'hatchling==1.18.0' hatch-vcs hatch-fancy-pypi-readme 'hatch-mypyc>=0.16.0' 'mypy==1.5.1' 'click==8.1.3'",
+    """sed -i '' -e "600,700s/'10_16'/os.environ['MACOSX_DEPLOYMENT_TARGET'].replace('.', '_')/" $(python -c 'import hatchling.builders.wheel as h; print(h.__file__)') """,
```

Comment:
the actual logic for this is https://github.com/pypa/wheel/blob/main/src/wheel/macosx_libfile.py#L392

Reply:
@ofek I think the thing to upstream would be this monkeypatch.

Reply:
We should have a way to set the sdk version in the tag to something based on `MACOSX_DEPLOYMENT_TARGET`, maybe this pseudocode:
```
sdk_ver = sdk_ver_from_narrowest_tag(tag)
if "MACOSX_DEPLOYMENT_TARGET" in os.environ:
    sdk_ver = tuple(map(int, os.environ["MACOSX_DEPLOYMENT_TARGET"].split(".")))
assert sdk_ver >= sysconfig.get_config_var("MACOSX_DEPLOYMENT_TARGET")
# then do equivalent of code from wheel^ to turn sdk_ver into a valid tag, can assert at the end that it is indeed in packaging's sys_tags
```